pymc/data.py

```diff
@@ -218,9 +218,9 @@ def Minibatch(variable: TensorVariable, *variables: TensorVariable, batch_size:
 def determine_coords(
     model,
     value: pd.DataFrame | pd.Series | xr.DataArray,
-    dims: Sequence[str | None] | None = None,
+    dims: Sequence[str] | None = None,
     coords: dict[str, Sequence | np.ndarray] | None = None,
-) -> tuple[dict[str, Sequence | np.ndarray], Sequence[str | None]]:
```

Comment:
Same as above. If variables can still have `dims=[None, "some_dim_name"]` then the old type hint is correct and the new one is wrong.

Reply:
I don't think we allow that anymore

Reply:
Based on [this](https://github.com/pymc-devs/pymc/pull/7473#discussion_r1793425912), the new type hint that you suggest is correct.